xdsl/printer.py

```diff
@@ -964,3 +964,7 @@ def print_metadata(self, dialects: Iterable[Dialect]) -> None:
 
         self.print_string("  }\n")
         self.print_string("#-}")
+
+    def print_symbol_name(self, sym_name: StringAttr):
+        self.print_string("@")
+        self.print_string(sym_name.data)
```

Comment:
Worth a separate test for new public function, IMO, as well as a doc string. OTOH I'm not entirely convinced that this is better than improving the SymbolNameAttr situation, which I think was originally meant for this purpose.

Reply:
I feel I agree, but am also unsure how the mechanism used by MLIR works